projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-notification-watcher.ts

```diff
@@ -0,0 +1,123 @@
+
+	useBadInstallNotice( redBubbleAlerts );
+	useSiteConnectionNotice( redBubbleAlerts );
+};
+
+export default useNotificationWatcher;
+
+const useSiteConnectionNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
+	const { setNotice } = useContext( NoticeContext );
+	const products = useAllProducts();
+	const navToConnection = useMyJetpackNavigate( MyJetpackRoutes.Connection );
+
+	useEffect( () => {
+		if ( ! Object.keys( redBubbleAlerts ).includes( 'missing-site-connection' ) ) {
+			return;
+		}
+
+		const productSlugsThatRequireUserConnection =
+			getProductSlugsThatRequireUserConnection( products );
+
+		const oneProductMessage = sprintf(
+			/* translators: placeholder is product name. */
+			__(
+				'Jetpack %s needs a user connection to WordPress.com to be able to work.',
+				'jetpack-my-jetpack'
+			),
+			productSlugsThatRequireUserConnection[ 0 ]
+		);
+
+		const needsUserConnectionMessage =
+			productSlugsThatRequireUserConnection.length === 1
+				? oneProductMessage
+				: __(
+						'Some products need a user connection to WordPress.com to be able to work.',
+						'jetpack-my-jetpack'
+				  );
+
+		const noticeOptions = {
+			status: 'warning',
+			actions: [
+				{
+					label: __( 'Connect your user account to fix this', 'jetpack-my-jetpack' ),
+					onClick: navToConnection,
+					noDefaultClasses: true,
+				},
+			],
+		};
+
+		setNotice( {
+			message: needsUserConnectionMessage,
+			options: noticeOptions,
+			priority: NOTICE_PRIORITY_HIGH,
+		} );
+	}, [ navToConnection, products, redBubbleAlerts, setNotice ] );
+};
+
+const useBadInstallNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
+	const { setNotice } = useContext( NoticeContext );
+	const { recordEvent } = useAnalytics();
+
+	useEffect( () => {
+		const badInstallAlerts = Object.keys( redBubbleAlerts ).filter( key =>
+			key.endsWith( '-bad-installation' )
+		) as Array< ` ${ string }-bad-installation` >;
+
+		if ( badInstallAlerts.length === 0 ) {
+			return;
+		}
+
+		const alert = redBubbleAlerts[ badInstallAlerts[ 0 ] ];
+		const { plugin } = alert.data;
+		const devEnvUrl =
+			'https://github.com/Automattic/jetpack/blob/trunk/docs/development-environment.md';
+
+		const errorMessagae = sprintf(
+			// translators: %s is the name of the plugin that has a bad installation.
+			__(
+				'Your installation of %1$s is incomplete. If you installed %1$s from GitHub, please refer to the developer documentation to set up your development environment.',
+				'jetpack-my-jetpack'
+			),
+			plugin
+		);
+
+		const onCtaClick = () => {
+			window.open( devEnvUrl );
+			recordEvent( 'jetpack_my_jetpack_bad_installation_notice_cta_click', {
+				plugin,
+			} );
+		};
+
+		const noticeOptions = {
+			status: 'error',
+			actions: [
+				{
+					label: __( 'See documentation', 'jetpack-my-jetpack' ),
+					onClick: onCtaClick,
+					noDefaultClasses: true,
+				},
+			],
+		};
+
+		setNotice( {
+			message: errorMessagae,
```

Comment:
Matching variable mis-spelling.

```suggestion
			message: errorMessage,
```